dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java

```diff
@@ -429,6 +431,62 @@ public Long execute(Jedis client, ConnectionContext state) {
         });
     }
 
+    @Override
+    public Object eval(String script, int keyCount, String... params) { return d_eval(script, keyCount, params).getResult(); }
+
+    public OperationResult<Object> d_eval(final String script, final int keyCount, final String... params) {
+        if (keyCount == 0) {
+            throw new DynoException("Need atleast one key in script");
+        }
+        return connPool.executeWithFailover(new BaseKeyOperation<Object>(params[0], OpName.EVAL) {
+            @Override
+            public Object execute(Jedis client, ConnectionContext state) {
+                return client.eval(script, keyCount, params);
+            }
+        });
+    }
+
+    @Override
+    public Object eval(String script, List<String> keys, List<String> args) {
+        String[] params = (String[])ArrayUtils.addAll(keys.toArray(), args.toArray());
+        return eval(script, keys.size(), params);
+    }
+
+    @Override
+    public Object eval(String script) {
+        return eval(script, 0);
+    }
+
+    @Override
+    public Object evalsha(String sha1, int keyCount, String... params) {
+        throw new UnsupportedOperationException("This function is Not Implemented. Please use eval instead.");
+    }
+
+    @Override
+    public Object evalsha(String sha1, List<String> keys, List<String> args) {
+        throw new UnsupportedOperationException("This function is Not Implemented. Please use eval instead.");
+    }
+
+    @Override
+    public Object evalsha(String script) {
+        throw new UnsupportedOperationException("This function is Not Implemented. Please use eval instead.");
+    }
+
+    @Override
+    public Boolean scriptExists(String sha1) {
```

Comment:
What is the issue and we cannot implement `scriptExists`?

Reply:
which node will you ask if a script exists? there is no key in a script till it is executed